crates/rpc/rpc.rs

```diff
@@ -208,10 +212,16 @@ pub fn map_requests(req: &RpcRequest, storage: Store) -> Result<Value, RpcErr> {
             let request = CallRequest::parse(&req.params).ok_or(RpcErr::BadParams)?;
             transaction::call(&request, storage)
         }
-        "engine_forkchoiceUpdatedV3" => engine::forkchoice_updated_v3(),
+        "engine_forkchoiceUpdatedV3" => {
+            let request = ForkChoiceUpdatedV3::parse(&req.params).ok_or(RpcErr::BadParams)?;
+            Ok(
+                serde_json::to_value(fork_choice::forkchoice_updated_v3(request, storage)?)
+                    .unwrap(),
+            )
+        }
         "engine_newPayloadV3" => {
             let request = NewPayloadV3Request::parse(&req.params).ok_or(RpcErr::BadParams)?;
-            Ok(serde_json::to_value(engine::new_payload_v3(request, storage)?).unwrap())
+            Ok(serde_json::to_value(payload::new_payload_v3(request, storage)?).unwrap())
```

Comment:
Could we return an error instead of unwrapping here?

Reply:
Done!